src/alchemlyb/tests/test_workflow_ABFE.py

```diff
@@ -388,62 +134,202 @@ def test_uncorr_threshold(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "u_nk_sample_list", [])
+        monkeypatch.setattr(workflow, "dHdl_sample_list", [])
         workflow.preprocess(threshold=50)
         assert len(workflow.u_nk_list) == 0
 
     def test_no_dHdl_preprocess(self, workflow, monkeypatch):
         monkeypatch.setattr(workflow, "dHdl_list", [])
+        monkeypatch.setattr(workflow, "u_nk_sample_list", [])
+        monkeypatch.setattr(workflow, "dHdl_sample_list", [])
         workflow.preprocess(threshold=50)
         assert len(workflow.dHdl_list) == 0
 
-    def test_single_estimator_mbar(self, workflow):
+
+class TestEstimator:
+    def test_default(self, workflow):
+        """Test if all three estimators have been used."""
+        assert len(workflow.estimator) == 3
+        assert "MBAR" in workflow.estimator
+        assert "TI" in workflow.estimator
+        assert "BAR" in workflow.estimator
+
+    def test_method(self, workflow, monkeypatch):
+        """Test if the method keyword could be passed to the AutoMBAR estimator."""
+        monkeypatch.setattr(workflow, "estimator", dict())
+        workflow.estimate(estimators="MBAR", method="adaptive")
+        assert "MBAR" in workflow.estimator
+
+    def test_single_estimator_ti(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "estimator", dict())
+        monkeypatch.setattr(workflow, "summary", None)
+        workflow.estimate(estimators="TI")
+        summary = workflow.generate_result()
+        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 2.946, 0.1)
+
+    def test_summary(self, workflow):
+        """Test if if the summary is right."""
+        summary = workflow.generate_result()
+        assert np.isclose(summary["MBAR"]["Stages"]["TOTAL"], 21.8, 0.1)
+        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 21.8, 0.1)
+        assert np.isclose(summary["BAR"]["Stages"]["TOTAL"], 21.8, 0.1)
+
+    def test_no_name_estimate(self, workflow):
+        with pytest.raises(ValueError):
+            workflow.estimate("aaa")
+
+    def test_single_estimator_mbar(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "estimator", dict())
         workflow.estimate(estimators="MBAR")
         assert len(workflow.estimator) == 1
         assert "MBAR" in workflow.estimator
         summary = workflow.generate_result()
-        assert np.isclose(summary["MBAR"]["Stages"]["TOTAL"], 2.946, 0.1)
+        assert np.isclose(summary["MBAR"]["Stages"]["TOTAL"], 21.645742066696315, 0.1)
 
-    def test_single_estimator_ti(self, workflow):
+    def test_single_estimator_ti(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "estimator", dict())
+        monkeypatch.setattr(workflow, "summary", None)
         workflow.estimate(estimators="TI")
         summary = workflow.generate_result()
-        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 2.946, 0.1)
+        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 21.51472826028906, 0.1)
```

Comment:
Why result change? Different input data?

At least TI and MBAR are agreeing...